tests/EntityControlTraitTest.php

```diff
@@ -2,33 +2,386 @@
 
 namespace RonasIT\Support\Tests;
 
-use RonasIT\Support\Repositories\BaseRepository;
+use Illuminate\Support\Carbon;
+use Mpyw\LaravelDatabaseMock\Facades\DBMock;
+use ReflectionMethod;
+use RonasIT\Support\Tests\Support\Mock\TestRepository;
 use RonasIT\Support\Tests\Support\Traits\MockTrait;
 use ReflectionProperty;
+use RonasIT\Support\Tests\Support\Traits\SqlMockTrait;
 
 class EntityControlTraitTest extends HelpersTestCase
 {
-    use MockTrait;
+    use MockTrait, SqlMockTrait;
 
-    protected BaseRepository $baseRepositoryClass;
+    protected TestRepository $testRepositoryClass;
     protected ReflectionProperty $onlyTrashedProperty;
+    protected ReflectionMethod $getQueryMethod;
+
+    protected array $selectResult;
 
     public function setUp(): void
     {
         parent::setUp();
 
-        $this->baseRepositoryClass = new BaseRepository();
+        $this->testRepositoryClass = new TestRepository();
 
-        $this->onlyTrashedProperty = new ReflectionProperty(BaseRepository::class, 'onlyTrashed');
+        $this->onlyTrashedProperty = new ReflectionProperty(TestRepository::class, 'onlyTrashed');
         $this->onlyTrashedProperty->setAccessible('pubic');
+
+        $this->getQueryMethod = new ReflectionMethod($this->testRepositoryClass, 'getQuery');
+        $this->getQueryMethod->setAccessible('public');
+
+        $this->selectResult = $this->getJsonFixture('select_query_result.json');
+
+        Carbon::setTestNow('2020-01-01 00:00:00');
     }
 
     public function testOnlyTrashed()
     {
-        $this->baseRepositoryClass->onlyTrashed();
+        $this->testRepositoryClass->onlyTrashed();
 
-        $onlyTrashed = $this->onlyTrashedProperty->getValue($this->baseRepositoryClass);
+        $onlyTrashed = $this->onlyTrashedProperty->getValue($this->testRepositoryClass);
 
         $this->assertEquals(true, $onlyTrashed);
     }
+
+    public function testGetQuery()
+    {
+        $sql = $this->getQueryMethod->invoke($this->testRepositoryClass)->toSql();
+
+        $this->assertEqualsFixture('get_query_sql.json', $sql);
+    }
+
+    public function testAll()
+    {
+        $pdo = DBMock::mockPdo();
+        $pdo
+            ->shouldSelect('select * from `test_models` where `test_models`.`deleted_at` is not null')
+            ->whenFetchAllCalled();
```

Comment:
let's move it to mock helper

```suggestion
        $this->mockSelect('select * from `test_models` where `test_models`.`deleted_at` is not null');
```